### ISSUE TITLE
Login Issues with known working account - Query email/password character issues?

### ISSUE THREAD

**Issue by danstreeter** [bdr99/pymazda#32]
Hey, I'm trying to use this eventually within Home Assistant, but dropped back to running it raw in Python to debug login issues without the HASS config wrapper potentially confusing things.

Getting persistent Invalid Credential (but now Account Locked) responses for simple login and vehicle list requests...

I've set up a new account, and successfully shared a vehicle from that account to it and can login via the App no problem on the 2nd account.
With either the Python or Node libraries I was getting the same results - `Invalid Email or Password` using the known working credential set I used on the app... even doing a reset to something simpler to avoid mistakes.

I then tried my original account with the node library; thinking it may be related to the TLS Fingerprinting issues within Python - and it logged in fine, reported vehicles back to me no problem. Changed the node library to the 2nd account - Invalid Credentials again.

Returned back to the Python library with 2nd account credentials and added tonnes of debug to the connection module to try and determine what the login request contains and yeilded no real additional information other than now the responses are coming back as `Account is locked` responses - however my App is still logged in fine on that 2nd account.
_(Even logging out of the app, then back in with the 2nd account works fine, making me think that the code isnt 'really' trying to log in with the actual 2nd account email address)_

This has led me to start thinking there is some discrepancy between what is being used to perform the login requests with this library between the credential set of my 1st / main & working in the node library and the 2nd account.

The differences are:
2nd set email is of the format `str.str@str.domain.co.uk`
1st set email is just `str@domain.co.uk`

2nd set password has `lower`, `upper`, `numerics` and a `special: $`
1st set password has `lower`, `upper`, `numerics` and a `special: !`

I'm wondering if something within that is causing the login request to be malformed somehow thus causing my login issue on only this 2nd account.

_Oh and just to round loop this thought - Using the 1st account credentials with the python library works fine... its literally something wrong with this set of credentials for the 2nd account which at 100% matched in App entry and this library entry._

Oh and forgot to add - happy to provide any debug assistance as required, just shout.

**Comment by bdr99** [comment]
Thanks for the issue report and detailed analysis! I'll try to reproduce this problem on my side and see if I can determine the cause.

**Comment by danstreeter** [comment]
@bdr99 I did some further tests last night and found that resetting the 2nd account to have a password the same as the first allowed it to login via the code successfullly.

I then further tried differing combinations of passwords and during the password change prompt found that the password is actually truncated within the app upon entry.

I did not notice as the password I was entering was only one character longer than the required amount, the app does not warn you that you went over, and just 'stops typing' in the box.

Reducing the password in the script by the last character now suceeds with the original (-1 char) password

**So no problem with the code here - apologies for the ticket raise**

Updated the component manually in my older version of HA and all is good and integration works as expected.
